### PR TITLE
Incorrect directory matching

### LINKED ISSUE #277: Incorrect directory matching in case of same directory name in the directory hierarchy
Continuation of: https://github.com/cqfn/save/pull/276/files#r719567913

### PR DESCRIPTION
**Pull request by Cheshiriks**
What's done:
* Fixed incorrect directory matching in case of same directory name in the directory hierarchy
Closes #277

### CODE REVIEW

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt [hunk 1]
```diff
@@ -91,7 +92,11 @@ abstract class Plugin(
             val foundTests = rawTestFiles.filter { rawTestFile ->
                 testFiles.any {
                     if (fs.exists(it.toPath())) {
-                        it in rawTestFile.test.toString()
+                        (it.toPath().parents().toList() + listOf(it.toPath())).all { testFile ->
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:95]: Pls add method `parentsWithSelf` or a parameter `parents(withSelf: Boolean = false)`

Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:95]:
```suggestion
                        (it.toPath().parents().toList() + listOf(it.toPath())).all { part ->
```
This argument name makes no sense, it implies that you are traversing `testFiles`, but you are not

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt [hunk 2]
```diff
@@ -91,7 +92,11 @@ abstract class Plugin(
...
+                            (rawTestFile.test.parents().toList() + listOf(rawTestFile.test)).any { rawTestFileDir ->
+                                testFile.toString() == rawTestFileDir.toString()
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:97]: Is this the logic we are aiming for, when they have at least one common parent directory?

**Review by petertrr — CHANGES_REQUESTED** [review]

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:97]: Not really. All parent directories of testFiles submitted by the user must match at least one parent directorie rawTestFiles

Comment (petertrr) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:97]: We can compare parents of `rawTestFile` with entire `testFile`

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt]: Fixed

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt]: Fixed

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt [hunk 3]
```diff
@@ -113,6 +111,8 @@ abstract class Plugin(
         }
     }
 
+    private fun Path.parentsWithSelf() = listOf(this) + this.parents().toList()
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:114]: Pls move to FileUtils

**Review by petertrr — APPROVED** [review]

**Pull request merged**